### DISCUSSION TITLE
Is it possible to lock layout spacing options to predetermined values?

### DISCUSSION THREAD

**Discussion by crs1138** [WordPress/gutenberg#42522] [Q&A] [answered]
How can I achieve a similar UI as for font sizes but with padding/margin values?
![image](https://user-images.githubusercontent.com/1313681/179699367-2d26542a-eeda-40c3-8fb9-ae26cda61891.png)

**Comment by fabiankaegy** [comment, marked as answer]
@crs1138 Not currently. But it is actively getting worked on in this issue here: https://github.com/WordPress/gutenberg/issues/39371

**Comment by crs1138** [comment]
I found `spacingSizes` in the `theme.json` schema. How do they work? Is there any example on how to use them?

https://developer.wordpress.org/block-editor/reference-guides/theme-json-reference/theme-json-living/

**Reply by fabiankaegy** [reply]
@crs1138 The Living documentation reflects the current state of the code in this GitHub repository. The `spacingSizes` key got added 19 days ago in this PR here: https://github.com/WordPress/gutenberg/pull/41527 and is part of the work that is outlined in the issue that I've shared initially. So it is still very much in progress. The current state is that the underlying architecture was started, but there is no UI for it yet and the implementation may change.

**Reply by crs1138** [reply]
Thanks for explaining. What `schema` value would you recommend for production ready themes? Let's say we're currently aiming to use the latest version of WP@6.0.1. How would I deal with many custom plugins and themes using a specific schema version when the time comes to upgrade the WP core?

**Reply by fabiankaegy** [reply]
If you want to use the latest version of the schema you can always use `https://schemas.wp.org/trunk/theme.json` which will give you the development living version which may include things that are not yet in core. 

For things that are in core you can use the release-specific branches like so:
- `https://schemas.wp.org/wp/5.8/theme.json`
- `https://schemas.wp.org/wp/5.9/theme.json`
- `https://schemas.wp.org/wp/6.0/theme.json`

The "major" version number always counts for the entire release cycle. So `6.0` will also pull any updates that happened in `6.0.1`

**Reply by fabiankaegy** [reply]
Essentially the way this works here is that the URL has a placeholder for the branch name here on GitHub in it:
`https://schemas.wp.org/${BRANCH_NAME}/theme.json`

**Reply by crs1138** [reply]
Would that be a branch of the @wordpress/gutenberg repo or @wordpress/wordpress?

**Reply by fabiankaegy** [reply]
@crs1138 the `@wordpress/gutenberg` repo

**Reply by fabiankaegy** [reply]
So for 5.9 for example the actual schema definition lives here: https://github.com/WordPress/gutenberg/blob/wp/5.9/schemas/json/theme.json

**Reply by crs1138** [reply]
Thank you for clarifying that.
